package-system/mikkelsen/build_package_image.py

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+# 
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import argparse
+import os
+import platform
+import subprocess
+import sys
+
+parser = argparse.ArgumentParser(description='Builds this package')
+parser.add_argument('--platform', default=platform.system().lower(), required=False, help=f'Platform to build (defaults to \"{platform.system().lower()}\")')
+args = parser.parse_args()
+
+this_dir = os.path.dirname(os.path.realpath(__file__))
+cmake_scripts_path = os.path.abspath(os.path.join(this_dir, '../../Scripts/cmake'))
+
+ly_3rdparty_path = os.getenv('LY_3RDPARTY_PATH')
+
+folder_names = { 
+    #system-name  cmake generation, cmake build
+    'mac'       : ([
+        '-G', 'Xcode'
+    ], []),
+    'ios'       : ([
+        '-G', 'Xcode',
+        f'-DCMAKE_TOOLCHAIN_FILE={cmake_scripts_path}/Platform/iOS/Toolchain_ios.cmake',
+        '-DPACKAGE_PLATFORM=ios'
+    ], [
+        '--',
+        '-destination generic/platform=iOS'
+    ]),
+    'linux'     : ([
+        '-G', 'Ninja Multi-Config',
+        '-DCMAKE_C_COMPILER=clang-6.0', 
+        '-DCMAKE_CXX_COMPILER=clang++-6.0'
+    ], []),
+    'windows'   : ([
+        '-G', 'Visual Studio 15 2017',
```

Comment:
We don't support VS2017 anymore, so we should change this to Visual Studio 16 at minimum.
Also for VS2019 the default architecture and host is x64, so the `-Ax64` and `Thost=x64` options aren't needed.

That being said I think we should data drive these options as much as possible.

Getting clang-6.0.0 on a modern Linux system that isn't Ubuntu 18.04 LTS is not easiest task.
Furthermore if the goal is to build the package on the user's system, then hard-coding clang-6.0 isn't really ideal when newer versions of clang are what available in the package system of the specific linux OS(i.e RedHat and ArchLinux comes with clang-12.0.0)

Being able to adjust the settings via a configuration file instead of a python script could increase the number of users that could build update the package one day.

Reply:
OK.   I'll file an issue to do this.  My current goal is to transplant the 3p libraries from the internal repos to somewhere useful from O3DE without incurring a whole lot of change during the process.